src/main/java/bio/terra/cli/service/WorkspaceManagerService.java

```diff
@@ -242,17 +243,34 @@ public WorkspaceDescription createWorkspace(
                   CREATE_WORKSPACE_MAXIMUM_RETRIES,
                   CREATE_WORKSPACE_DURATION_SLEEP_FOR_RETRY);
           logger.debug("create workspace context result: {}", createContextResult);
-
-          // if this is a spend profile access denied error, then throw a more user-friendly error
-          // message
-          if (createContextResult.getJobReport().getStatus().equals(JobReport.StatusEnum.FAILED)
-              && createContextResult.getErrorReport().getMessage().contains("spend profile")
-              && createContextResult.getErrorReport().getStatusCode()
-                  == HttpStatusCodes.STATUS_CODE_FORBIDDEN) {
-            throw new UserActionableException(
-                "Accessing the spend profile failed. Ask an administrator to grant you access.");
+          StatusEnum status = createContextResult.getJobReport().getStatus();
+          if (StatusEnum.FAILED == status) {
+            // need to delete the empty workspace before continuing
+            boolean workspaceSuccessfullyDeleted = false;
+            try {
+              deleteWorkspace(workspaceId);
+              workspaceSuccessfullyDeleted = true;
+            } catch (SystemException ex) {
+              logger.error("Failed to delete workspace {} when cleaning up failed creation of cloud context. "
```

Comment:
`logger.error` optionally takes a `Throwable` as its last argument, which will also log the stack trace. This is helpful in case the message is not helpful or undefined.

Reply:
good call